test/e2e/integration/utils.go

```diff
@@ -76,18 +88,20 @@ func (cmd *Cmd) RunCmdWithFlags() error {
 				break
 			}
 			if len(line) > 0 {
-				logWriter.WriteString(fmt.Sprintf("%s\n", line))
-				logWriter.Flush()
+				if _, err = logWriter.WriteString(fmt.Sprintf("%s\n", line)); err != nil {
+					cmd.Logger.Errorf("failed to write to log file for command %s: %v", cmd.Task, err)
+				}
+				if err = logWriter.Flush(); err != nil {
+					cmd.Logger.Errorf("failed to flush log file for command %s: %v", cmd.Task, err)
+				}
 			}
 		}
 	}()
 
-	if err := cmd.process.Wait(); err != nil {
-		if exiterr, ok := err.(*exec.ExitError); ok {
-			cmd.Logger.Infof("command execution stopped (signal: %s)", exiterr.Error())
-		} else if err != nil {
-			cmd.Logger.Errorf("command %s failed with error: %v", cmd.Task, err)
-			return err
+	if err = cmd.process.Wait(); err != nil {
+		var exitErr *exec.ExitError
+		if errors.As(err, &exitErr) {
+			cmd.Logger.Infof("command execution stopped (signal: %s)", exitErr.Error())
```

Comment:
do you want to print this log as info ?

Reply:
I didn't change anything here, just changed the error parsing on line 103. I have no strong opinion on how we print the log line here, since it's just a test and will be printed anyway

Reply:
I think error log would make more sense while debugging in case of e2e tests failure.